### PR TITLE
feat(controller): add recentlyUsed flag for artifact vo

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description
Add recentlyUsed(top 5) flag for artifact view vo.

> recentlyUsed artifact's defintion: the artifact which was used in job recently

cc @waynelwz 

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Review by anda-ren — DISMISSED** [review]
LGTM

**Comment by anda-ren** [comment]
Maybe we need a performance analysis against the SQL in the future

File: server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/RuntimeService.java [hunk 1]
```diff
@@ -225,6 +225,18 @@ public List<RuntimeViewVo> listRuntimeVersionView(
...
+            var recentlyUsed = runtimeVersionMapper.listRuntimeVersionsByUserRecentlyUsed(projectId, userId, 5)
+                    .stream().map(version -> String.valueOf(version.getId())).collect(Collectors.toList());
+            for (var vo : list) {
+                for (var version : vo.getVersions()) {
+                    if (recentlyUsed.contains(version.getId())) {
+                        version.setRecentlyUsed(true);
+                    }
+                }
+            }
+        }
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/RuntimeService.java:239]: Can this page have no recent items?
Or essentially, we need to order by recent usage, right?

**Comment by goldenxinxing** [comment]
> Maybe we need a performance analysis against the SQL in the future

OK，This is the explain for sql which belong to model:
image
only the table of model_info is all,  I think the number of artifact's Info table should not be too large.

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**